src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -196,13 +220,13 @@ public Optional<Name> getName() {
             return Optional.ofNullable(name);
         }
 
-        public void setCategory(String category) {
-            this.category = category;
-        }
+        public void setStudentID(StudentID studentID) { this.studentID = studentID; }
 
-        public String getCategory() {
-            return category;
-        }
+        public Optional<StudentID> getStudentID() { return Optional.ofNullable(studentID); }
```

Comment:
good use of the Optional class to indicate that the StudentId field could be empty